src/core/ad.jl

```diff
@@ -57,7 +58,8 @@ function gradient_reverse(
   # Specify objective function.
   function f(θ)
     vi[spl] = θ
-    return -runmodel(model, vi, spl).logp
+    runmodel(model, vi, spl)
+    return -vi.logp
```

Comment:
What's the rationale for splitting this up over two lines?

Reply:
Calling the model function does not guarantee to return a VarInfo object anymore. Would it be better to change the behaviour of the runmodel function so that we still get a VarInfo object as the return value?

Reply:
Oh I see. Yes, I would be in favour of that. While you're at it could you also please rename `runmodel` -> `runmodel!`, as it definitely mutates it `VarInfo` argument.